seaborn/categorical.py

```diff
@@ -1797,78 +1799,72 @@ def __init__(self, x, y, hue, data, order, hue_order,
         self.k_depth = k_depth
 
+        # TODO seems this member is only used to frame the legend artists
         if linewidth is None:
             linewidth = mpl.rcParams["lines.linewidth"]
         self.linewidth = linewidth
 
-        if scale is None:
-            scale = 'exponential'
+        if scale not in ('linear', 'exponential', 'area'):
+            raise ValueError(f'Scale "{scale}" not recognized')
         self.scale = scale
 
+        if ((outlier_prop > 1) or (outlier_prop < 0)):
+            msg = f'outlier_prop {outlier_prop} not in range [0, 1]'
+            raise ValueError(msg)
         self.outlier_prop = outlier_prop
 
         self.showfliers = showfliers
 
         self.establish_variables(x, y, hue, data, orient, order, hue_order)
         self.establish_colors(color, palette, saturation)
 
-    def _lv_box_ends(self, vals, k_depth='proportion', outlier_prop=None):
+    def _lv_box_ends(self, vals):
         """Get the number of data points and calculate `depth` of
         letter-value plot."""
         vals = np.asarray(vals)
         vals = vals[np.isfinite(vals)]
         n = len(vals)
-        # If p is not set, calculate it so that 8 points are outliers
-        if not outlier_prop:
-            # Conventional boxplots assume this proportion of the data are
-            # outliers.
-            p = 0.007
-        else:
-            if ((outlier_prop > 1.) or (outlier_prop < 0.)):
-                raise ValueError('outlier_prop not in range [0, 1]!')
-            p = outlier_prop
+        p = self.outlier_prop
+
         # Select the depth, i.e. number of boxes to draw, based on the method
-        k_dict = {
-            'proportion': (np.log2(n)) - int(np.log2(n * p)) + 1,
-            'tukey': (np.log2(n)) - 3,
-            'trustworthy': 1 + (np.log2(n)
-                                - np.log2(2 * stats.norm.ppf((1 - p)) ** 2))
-        }
-        k = k_dict[k_depth]
-        try:
-            k = int(k)
-        except ValueError:
-            k = 1
-        # If the number happens to be less than 0, set k to 0
-        if k < 1.:
+        if p == 0:
+            # Zero outliers; extend boxes to 100% of the data
+            k = int(np.log2(n)) + 1
+        elif self.k_depth == 'tukey':
+            # This results with 5-8 points in each tail
+            k = int(np.log2(n)) - 3
+        elif self.k_depth == 'proportion':
+            k = int(np.log2(n)) - int(np.log2(n * p)) + 1
+        elif self.k_depth == 'trustworthy':
+            k = int(np.log2(n / (2 * stats.norm.ppf((1 - p)) ** 2))) + 1
+        else:
+            k = int(self.k_depth)  # allow having k as input
+        # If the number happens to be less than 1, set k to 1
+        if k < 1:
             k = 1
+
         # Calculate the upper box ends
-        upper = [100 * (1 - 0.5 ** (i + 2)) for i in range(k, -1, -1)]
+        upper = [100 * (1 - 0.5 ** (i + 1)) for i in range(k, 0, -1)]
```

Comment:
Is there any reason not to add 1 to `k` before doing this math and the math in `_lv_outliers`? (while changing `range(k, 1, -1)`?

Reply:
For this enumeration, it seems more natural to use `k` since it refers to the number of boxes; the outer edge of box `i` is the beginning of box `i+1` hence the `i+1`. I will note this in the comment above this line. For `_lv_outliers` I will add 1 to k.